src/Elmish.WPF.Tests/DynamicViewModelTests.fs

```diff
@@ -1410,6 +1410,75 @@ module SubModelSelectedItem =
 
 
 
+module CacheEffect =
+
+  [<Fact>]
+  let ``model mapping called excactly once when Get called twice`` () =
+
+    Property.check <| property {
+      let! name = GenX.auto<string>
+      let! model = GenX.auto<int>
+      let! bindingComponsitionOrder = Gen.bool
+    
+      let mapping = InvokeTester id
+      let cachingAndMapping =
+        if bindingComponsitionOrder
+        then Binding.mapModel mapping.Fn >> Binding.addCaching
+        else Binding.addCaching >> Binding.mapModel mapping.Fn
+      let binding =
+        name
+        |> Binding.OneWay.id
+        |> cachingAndMapping
+      let vm = TestVm(model, binding)
+      
+      vm.Get name |> ignore // populate cache
+      vm.Get name |> ignore
+      
+      test <@ 1 = mapping.Count @>
+    }
+
+
+  [<Fact>]
+  let ``second Get returns new model after first Get and then Update`` () =
+    let name = ""
+    let model = 0
+    let newModel = 1
+    let binding =
+      name
+      |> Binding.OneWay.id
+      |> Binding.addCaching
+    let vm = TestVm(model, binding)
+
+    vm.Get name |> ignore   // populate cache
+    vm.UpdateModel newModel // clear cache
+    let actual = vm.Get name |> unbox
+
+    test <@ newModel = actual @>
+
+
+  [<Fact>]
+  let ``cache not cleared on Set`` () =
+    let name = ""
+    let initialModel = 0
+    let newModel = 1
+    let mapping = InvokeTester (fun x -> x)
+    let binding =
+      name
+      |> Binding.TwoWay.id
+      |> Binding.mapModel mapping.Fn
+      |> Binding.addCaching
+    let vm = TestVm(initialModel, binding)
+
+    vm.Get name |> ignore // populate cache
+    vm.Set name newModel
+    mapping.Reset()       // Set calls mapping function, so reset count
```

Comment:
@marner2, I don't think this caching effect can prevent the model mapping function from being called on `Set` when the cache is populated.  `Set` needs the model but the cache contains a different type.

Do you agree?

Reply:
I'm fighting very hard against this, but I think you're right. "Set" is really "DispatchMessage." The model isn't affected until you actually run "Update."

I agree.

Reply:
Can you elaborate on what you mean by

> I'm fighting very hard against this

?

Reply:
It just seems wrong to not bust the cache when you set it. But I think I'm pretty convinced that the "actual" set happens on `Update`.

Reply:
Maybe as part of another PR. I think it’s clear enough here.